packages/jaeger-ui/src/model/span.tsx

```diff
@@ -24,3 +25,89 @@ export function getParent(span: Span) {
   const parentRef = span.references ? span.references.find(ref => ref.refType === 'CHILD_OF') : null;
   return parentRef ? parentRef.span : null;
 }
+
+/** Shape of data required to determine a span's group. */
+type SpanGroupData = {
+  process: {
+    serviceName: string;
+    tags: Array<KeyValuePair>;
+  };
+  tags: Array<KeyValuePair>;
+};
+
+/**
+ * Generates a group for the span that can be used when determining which
+ * colour it should have, how many distinct groups of spans exist, and so on.
+ *
+ * The group is determined from the `spanGroupKey` configuration object, with
+ * properties:
+ *
+ * - `preset` (string) - set the fields according to presets designed to work with
+ *   conventions adopted by different projects.
+ *
+ * - `tags` (string[]) - set the group according to the values of the tags
+ *   listed in this array. Process tags are searched first, followed by span
+ *   tags. The special value `serviceName` corresponds to the service's name.
+ *
+ * If `preset` is specified then `tags` is ignored.
+ *
+ * # Available presets
+ *
+ * - {@link https://github.com/open-telemetry/opentelemetry-specification/blob/ef4612dd0b1f4266e6eca7177290ee2a5f087199/specification/resource/semantic_conventions/README.md|otel-ef4612d},
```

Comment:
a) for better readability of the doc, I would move the link to details and explicitly list the actual constant first, e.g.
```
 * - "otel-xxx" - Equivalent to a combination of tags: ['serviceName', 'service.namespace', 'service.instance.id]`.
 *    @see [OTEL semantic conventions](...link...)
b) `otel-ef4612d` is a confusing present name, and not descriptive of the preset itself, I suggest `otel-service-instance`
c) it would be better to define the default as a preset too, e.g. `service-name`